hibernate-architecture/README.md

```diff
@@ -0,0 +1,117 @@
+# Hibernate Architecture
+> 들어가면서..
+- 현재 Spring 5, Hibernate 5, HikariCP 사용 중.
+- 목표는? database-routing (main <-> repl)
+    - 검색해보면?
+    - https://www.baeldung.com/spring-abstract-routing-data-source 와 같은 방법을 찾아볼 수도 있음.
+- 그렇지만 현재 환경에서 안될 수도?
+    - 왜? 
+    - Dao Layer 에서 SessionFactory를 직접 주입 받아서 하고 있기 때문에
+    - 위 포스팅은 `Datasource` 레벨에서의 스위칭 인듯
+- 그러면.. 만약에 A- TransactinoManager가 진행 중인 트랜잭션 중에 B- SessionFactory로 switch 한다면, 어떻게 될까?
+    - 에러? 성공?..
+
+
+## Architecture 
+![](https://access.redhat.com/webassets/avalon/d/Red_Hat_JBoss_Web_Server-3-Hibernate_Core_Reference_Guide-en-US/images/c3ee197b49364a876bc5867d5c2c6db7/1091.png)
+
+
+### SessionFactory
+- Session 을 생성하는 공장, (그만큼 이 객체를 생성하는 데 많은 자원이 소모되는 듯.)
```

Comment:
> When we are using more than one database in our application than we use the HibernateUtil class which is implemented based on singleton design pattern which insures that one and only one sessionFactory object will be created for entire application.Session factory objects are to be implemented using the singleton design pattern. Instances of SessionFactory are thread-safe and typically shared throughout an application. Because creation of a SessionFactory is an extremely expensive process which involves parsing hibernate configuration/mapping properties and creating database connection pool.

[원문](http://javarerevisited.blogspot.com/2015/01/why-only-one-sessionfactory-object-per.html)